Entirely restructure analyzer for better readability & future extensibility

WizardCM opened the pull request:
### Description

* Create subdirectory for HTML templates
* Split most functions out of loganalyzer.py into checks/
* Convert log file regex checks to reusable functions
* Added requirements.txt to document dependencies
* Added setup.cfg to configure rules for pycodestyle
* All code formatted with pycodestyle
* Improved Readme documentation

This PR does not change any core functionality, just shuffles code around and simplifies the readability.

I'm submitting this as a PR to get a little extra testing from others, and to get general feedback on filenames & folder structure.

New structure:
```
├── check_pep8.sh
├── checks
│   ├── utils
│   │   ├── __init__.py
│   │   ├── fetchers.py
│   │   ├── utils.py
│   │   └── windowsversions.py
│   ├── __init__.py
│   ├── audio.py   
│   ├── core.py    
│   ├── encoding.py
│   ├── graphics.py
│   ├── network.py 
│   ├── plugins.py 
│   ├── sources.py
│   ├── vars.py
│   └── windows.py
├── LICENSE
├── loganalyzer.py
├── README.md
├── simplehttp.py
└── templates
    ├── detail.html
    └── index.html
```

### Motivation and Context

The log analyzer has become a fundamental tool for support volunteers & general troubleshooting. In its current state, it can be hard to navigate, and can scare off potential contributors due to its sheer size.

### How Has This Been Tested?

Run a variety of logs from various sources via the command line & web API.

### Types of changes

- Code cleanup (non-breaking change which makes code smaller or more readable)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

tt2468 commented:
All looks good to me. When this gets merged I can make a PR to convert the bot to using asyncio and some more modern HTTP stuff. Do you happen to have a test setup?

WizardCM commented:
After a little extra work, I've now added a `requirements.txt` file to make setup easier, a `setup.cfg` with settings for `pycodestyle`, formatted the code, and added setup instructions to the Readme.

WizardCM commented:
OK, Final push done. Added two GitHub Actions runs. One to test a fairly noisy log file (one of my own), and one to check that the code is formatted. Run details available here: https://github.com/WizardCM/loganalyzer/actions/runs/779577684

![image](https://user-images.githubusercontent.com/941350/115944238-ed4c2580-a4f7-11eb-83d9-8c4f9dfa33c9.png)
